packages/wonder-blocks-labeled-field/src/components/labeled-field.tsx

```diff
@@ -1,4 +1,169 @@
-// Something so that the file is not empty
-const LabeledField = "LabeledField";
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
 
-export default LabeledField;
+import {View, addStyle, StyleType} from "@khanacademy/wonder-blocks-core";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import {color, spacing} from "@khanacademy/wonder-blocks-tokens";
+import {LabelMedium, LabelSmall} from "@khanacademy/wonder-blocks-typography";
+
+type Props = {
+    /**
+     * The form field component.
+     */
+    field: React.ReactNode;
+    /**
+     * The title for the label element.
+     */
+    label: React.ReactNode;
+    /**
+     * The text for the description element.
+     */
+    description?: React.ReactNode;
+    /**
+     * Whether this field is required to continue.
+     */
+    required?: boolean;
+    /**
+     * The message for the error element.
+     */
+    error?: React.ReactNode;
+    /**
+     * Custom styles for the labeled field container.
+     */
+    style?: StyleType;
+    /**
+     * A unique id to link the label (and optional error) to the field.
+     *
+     * The label will assume that the field will have its id formatted as `${id}-field`.
+     * The field can assume that the error will have its id formatted as `${id}-error`.
+     */
+    id?: string;
+    /**
+     * Optional test ID for e2e testing.
+     */
+    testId?: string;
+    /**
+     * Change the field’s sub-components to fit a dark background.
+     */
+    light?: boolean;
+};
+
+const StyledSpan = addStyle("span");
+
+/**
+ * A LabeledField is an element that provides a label, description, and error element
+ * to present better context and hints to any type of form field component.
+ */
+export default class LabeledField extends React.Component<Props> {
```

Comment:
**suggestion:** Probably out of scope for this PR, but it would be nice if this could be refactored to a function component, so you can start using hooks immediately in future PRs.

Reply:
Good idea! In the next PR, I'll refactor this to a function component 👍